src/init/rosella_features.rs

```diff
@@ -0,0 +1,308 @@
+                self
+            }
+
+            fn as_any_mut(&mut self) -> &mut dyn Any {
+                self
+            }
+        }
+    }
+}
+
+/// Instance feature which provides all requirements needed for rosella to function in headless
+#[derive(Default)]
+pub struct RosellaInstanceBase;
+const_instance_feature!(RosellaInstanceBase, "rosella:instance_base", [KHRTimelineSemaphore::NAME]);
+
+impl ApplicationInstanceFeature for RosellaInstanceBase {
+    fn init(&mut self, features: &mut dyn FeatureAccess, _: &InstanceInfo) -> InitResult {
+        if !features.is_supported(&KHRTimelineSemaphore::NAME.get_uuid()) {
+            log::warn!("KHRTimelineSemaphore is not supported");
+            return InitResult::Disable;
+        }
+
+        InitResult::Ok
+    }
+
+    fn enable(&mut self, _: &mut dyn FeatureAccess, _: &InstanceInfo, _: &mut InstanceConfigurator) {
+    }
+}
+
+/// Instance feature which loads validation layers and provides debug callback logging
+#[derive(Default)]
+pub struct RosellaDebug;
+const_instance_feature!(RosellaDebug, "rosella:instance_debug", []);
+
+impl RosellaDebug {
+    extern "system" fn debug_callback(severity: vk::DebugUtilsMessageSeverityFlagsEXT, _: vk::DebugUtilsMessageTypeFlagsEXT, data:*const vk::DebugUtilsMessengerCallbackDataEXT, _:*mut c_void) -> vk::Bool32 {
+        let data = unsafe { data.as_ref().unwrap() };
+
+        let id = match unsafe { CStr::from_ptr(data.p_message_id_name) }.to_str() {
+            Ok(str) => str,
+            Err(err) => {
+                log::error!("Failed to process debug callback id: {:?}", err);
+                return vk::FALSE;
+            }
+        };
+
+        let msg = match unsafe { CStr::from_ptr(data.p_message) }.to_str() {
+            Ok(str) => str,
+            Err(err) => {
+                log::error!("Failed to process debug callback message: {:?}", err);
+                return vk::FALSE;
+            }
+        };
+
+        if severity.contains(vk::DebugUtilsMessageSeverityFlagsEXT::ERROR) {
+            log::error!(target: "vulkan", "{}: {}", id, msg);
+        } else if severity.contains(vk::DebugUtilsMessageSeverityFlagsEXT::WARNING) {
+            log::warn!(target: "vulkan", "{}: {}", id, msg);
+        } else if severity.contains(vk::DebugUtilsMessageSeverityFlagsEXT::INFO) {
+            log::info!(target: "vulkan", "{}: {}", id, msg);
+        } else {
+            log::debug!(target: "vulkan", "{}: {}", id, msg);
+        }
+
+        vk::FALSE
+    }
+}
+
+impl ApplicationInstanceFeature for RosellaDebug {
+    fn init(&mut self, _: &mut dyn FeatureAccess, info: &InstanceInfo) -> InitResult {
+        if !info.is_extension_supported::<ash::extensions::ext::DebugUtils>() {
+            log::warn!("DebugUtils extension not found! Rosella debug will be disabled.");
+            return InitResult::Disable;
+        }
+
+        if !info.is_layer_supported_str("VK_LAYER_KHRONOS_validation") {
```

Comment:
i thought it was `VK_LAYER_KHRONOS_VALIDATION`?

Reply:
Took this straight from their github page https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/docs/khronos_validation_layer.md

Also i tested the code and its properly loading the layer,